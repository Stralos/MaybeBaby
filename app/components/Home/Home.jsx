import React, { Component } from "react";
import { Link } from "react-router-dom";
import {
  Title,
  Header
} from "./Styles";

type Props        = { greeting: string, date: number };
type State        = { greeting: string };


export default class Main extends Component<void, Props, State> {

  state = {
    greeting: this.props.greeting,
  }

  onClick = () => {
    console.log("Hello World!");
  }

  render(){
    const { greeting } = this.state;
    return (
      <div>
        <Title>
          I am a title : )
        </Title>
       <Header>
        I am header !
       </Header>
        { greeting }
        { this.props.date}
        <button onClick={this.onClick}> Click Me!</button>
        <Link to="/services"> Services!</Link>
        <Link to="/contacts"> Contacts!</Link>
      </div>
    )
  }
}