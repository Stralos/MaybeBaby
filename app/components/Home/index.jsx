/* @flow */
import React, { Component } from "react";

type Props        = { greeting: string, date: number };
type State        = { greeting: string };

export default class Main extends Component<void, Props, State> {

  state = {
    greeting: this.props.greeting,
  }

  onClick = () => {
    console.log("Hello World!");
  }

  render(){
    const { greeting } = this.state;
    return (
      <div>
        { greeting }
        { this.props.date}
        <button onClick={this.onClick}> Click Me!</button>
      </div>
    )
  }
}