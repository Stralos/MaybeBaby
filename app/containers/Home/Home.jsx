/* @flow */
import React, { PureComponent } from 'react';
import styled from 'styled-components';

import WorkTime from './__components__/WorkTime';

export type Props = {
  +image: string,
  +description: string,
  +products: Array<{|
    +image: string,
  |}>,
  +workingHours: Array<{
    +day: string,
    +open: boolean,
    +opensAt: string,
    +closesAt: string,
  }>,
  +address: string,
  +phone: string,
  +email: string,
}

export const Description = styled.div`
  margin-top: 40px;
  margin-bottom: 40px;
`;

export const ProductList = styled.ul`
  padding: 0;
  margin: 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-bottom: 40px;
  margin: 0px -12px 40px -12px;
`;

export const Product = styled.li`
  height: 101px;
  background-color: #f2f2f2;
  list-style-type: none;
  margin: 0 12px 0 12px;
  flex: 2;
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const ProductImage = styled.img`
  filter: grayscale(1);
  max-width: calc(100% - 30px);
  max-height: calc(100% - 26px);
`;

export const Contacts = styled.div`
  display: flex;
`;
export const ContactDetails = styled.div`
`;
export const Phone = styled.div`
`;
export const Address = styled.div`
`;
export const Email = styled.div`
`;
export const TimeTable = styled.ul`
  li {
    margin: 2px 0;
  }
`;
export const Header = styled.h3`
  display: flex;
  justify-content: center;
  font-size: 1.75rem;
  margin-bottom: 24px;
  font-family: 'Marcellus SC';
`;
export const VenueImage = styled.img`
  max-width: 100%;
  height: auto;
  display: block;
  margin: 0 auto;
`;

export default class Home extends PureComponent<Props> {
  renderProductList = () => {
    const { products } = this.props;
    const productList = products.map((product) => {
      const { image } = product;
      return (
        <Product key={image}>
          <ProductImage src={image} />
        </Product>
      );
    });

    return (
      <div>
        <Header>
          Products we use
        </Header>
        <ProductList>
          { productList }
        </ProductList>
      </div>
    );
  }

  renderTimeTable = () => {
    const { workingHours } = this.props;
    return (
      <TimeTable>
        { workingHours.map(time => <WorkTime {...time} key={time.day} />)}
      </TimeTable>
    );
  }

  renderContactDetails = () => {
    const { address, email, phone } = this.props;
    return (
      <ContactDetails>
        <Address>
          { address }
        </Address>
        <Phone>
          { phone }
        </Phone>
        <Email>
          { email }
        </Email>
      </ContactDetails>
    );
  }

  renderContacts = () => {
    return (
      <div>
        <Header>
          Contact
        </Header>
        <Contacts>
          { this.renderContactDetails() }
          { this.renderTimeTable() }
        </Contacts>
      </div>
    );
  }

  render() {
    const { description, image } = this.props;
    return (
      <div>
        <VenueImage src={image} align="middle" />
        <Description>
          { description }
        </Description>
        {this.renderProductList()}
        {this.renderContacts()}
      </div>
    );
  }
}
