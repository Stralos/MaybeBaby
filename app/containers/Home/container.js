/* @flow */
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import type { Connector } from 'react-redux';

import Home from './Home';
import type { Props } from './Home';

type OwnProps = {
  +route: string,
}

const mapSateToProps = state => Object.assign({}, {
  greeting: state.greeting,
  date: state.date,
});

const connector : Connector<OwnProps, Props> = connect(mapSateToProps, {});

export default withRouter(connector(Home));
