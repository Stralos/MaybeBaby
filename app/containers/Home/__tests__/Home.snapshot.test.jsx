/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Home from '../Home';
import type { Props } from '../Home';

const getProps = (): Props => ({
  image: 'www.images.com/venueImage.jpg',
  description: 'This is the best venue Ever!',
  products: [{
    image: 'www.images.com/shampoon.jpg',
  }, {
    image: 'www.images.com/soap.jpg',
  }],
  workingHours: [{
    day: 'Monday',
    open: false,
    opensAt: '08:00',
    closesAt: '18:00',
  }, {
    day: 'Tuesday',
    open: true,
    opensAt: '10:00',
    closesAt: '20:00',
  }],
  address: 'England London Main Street 13',
  phone: '+88 2349 83921',
  email: 'awesomeVenue@gmail.com',
});


describe('Home Component Snapshot-- ', () => {
  it('should renders correctly', () => {
    const props = getProps();
    const tree = renderer
      .create(<Home {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
