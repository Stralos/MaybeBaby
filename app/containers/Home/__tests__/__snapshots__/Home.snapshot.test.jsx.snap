// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Snapshot--  should renders correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.75rem;
  margin-bottom: 24px;
  font-family: 'Marcellus SC';
}

.c9 {
  padding-left: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 28.8px;
}

.c9:before {
  top: 50%;
  left: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  position: absolute;
  height: 10px;
  width: 10px;
  content: '';
  border-radius: 50%;
  border: 1px solid white;
  box-sizing: border-box;
  background-color: transparent;
}

.c10 {
  padding-left: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 28.8px;
}

.c10:before {
  top: 50%;
  left: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  position: absolute;
  height: 10px;
  width: 10px;
  content: '';
  border-radius: 50%;
  border: 1px solid white;
  box-sizing: border-box;
  background-color: white;
}

.c1 {
  margin-top: 40px;
  margin-bottom: 40px;
}

.c3 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 40px;
  margin: 0px -12px 40px -12px;
}

.c4 {
  height: 101px;
  background-color: #f2f2f2;
  list-style-type: none;
  margin: 0 12px 0 12px;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
  max-width: calc(100% - 30px);
  max-height: calc(100% - 26px);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c8 li {
  margin: 2px 0;
}

.c0 {
  max-width: 100%;
  height: auto;
  display: block;
  margin: 0 auto;
}

<div>
  <img
    className="c0"
    src="www.images.com/venueImage.jpg"
  />
  <div
    className="c1"
  >
    This is the best venue Ever!
  </div>
  <div>
    <h3
      className="c2"
    >
      Products we use
    </h3>
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        <img
          className="c5"
          src="www.images.com/shampoon.jpg"
        />
      </li>
      <li
        className="c4"
      >
        <img
          className="c5"
          src="www.images.com/soap.jpg"
        />
      </li>
    </ul>
  </div>
  <div>
    <h3
      className="c2"
    >
      Contact
    </h3>
    <div
      className="c6"
    >
      <div
        className="c7"
      >
        <div>
          <div
            className=""
          >
            Address
          </div>
          <div
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "England London Main Street 13",
              }
            }
          />
        </div>
        <div>
          <div
            className=""
          >
            Phone
          </div>
          <div
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "+88 2349 83921",
              }
            }
          />
        </div>
        <div>
          <div
            className=""
          >
            Email
          </div>
          <div
            className=""
            dangerouslySetInnerHTML={
              Object {
                "__html": "awesomeVenue@gmail.com",
              }
            }
          />
        </div>
      </div>
      <ul
        className="c8"
      >
        <li
          className="c9"
          open={false}
        >
          <span
            className=""
          >
            Monday
          </span>
          <span
            className=""
          >
            08:00 - 18:00
          </span>
        </li>
        <li
          className="c10"
          open={true}
        >
          <span
            className=""
          >
            Tuesday
          </span>
          <span
            className=""
          >
            10:00 - 20:00
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
